src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplate.cs

```diff
@@ -84,54 +87,54 @@ public static BindingTemplate FromString(string input, bool ignoreCase = false)
         /// <returns>Resolved string if succeeded.</returns>
         /// <exception cref="InvalidOperationException">Thrown when required parameter value is not available.
         /// </exception>
-        public string Bind(IReadOnlyDictionary<string, string> parameters)
+        public string Bind(IReadOnlyDictionary<string, object> parameters)
         {
             StringBuilder builder = new StringBuilder();
 
             if (_ignoreCase && parameters != null)
             {
                 // convert to a case insensitive dictionary
-                var caseInsensitive = new Dictionary<string, string>(parameters.Count, StringComparer.OrdinalIgnoreCase);
+                var caseInsensitive = new Dictionary<string, object>(parameters.Count, StringComparer.OrdinalIgnoreCase);
```

Comment:
nit: Linq ToDictionary reads a bit cleaner to me, same below. 

Also, I don't like that we have at least 3 separate dictionaries created if <string, string> is used, and at least 2 for <string, object> - if we provided a different api to the extension which added individual binding data values instead of one dictionary we could remove all these conversions